### ISSUE TITLE
Possible NPE in exchangeAuthorizationCodeFor2FAResult

### ISSUE THREAD

**Issue by philsmart** [duosecurity/duo_universal_java#8]
Line 338 of the Client class e.g.
```
String idToken = response.getId_token();
```
Causes an NPE if (in the unlikely event) the auth code is invalid or not present. The Token endpoint correctly returns an HTTP 400, so I think the logic just needs to check the response is actually present and correct before trying to access the id_token.

**Comment by hkg1990** [comment]
I observed similar issue. It seems that happened when Remembered-Device feature is utilized.

**Comment by nie7321** [comment]
I'm hitting this NPE with some of my integrations. They're getting a 401 Unauthorized response, with this body:

```json
{"error": "invalid_client", "error_description": "Customer does not have frameless access enabled."}
```

For whatever reason, this isn't throwing a Duo exception but instead returns a `null` response.

I don't know how to reliably reproduce that -- it only seems to affect some of my integrations, not all of them.

**Comment by DuoKristina** [comment]
@nie7321 That's definitely not expected as Duo's frameless Web SDK was GA last October.

Are you using this new SDK with an application that supports the frameless prompt (like the one called "Web SDK" or another from the list [here](https://duo.com/docs/universal-prompt-update-guide#application-update-instructions)? When you look at an application that throws this 401 in the Duo Admin Panel, does the Universal Prompt part of the page show that the status is either "App Update Ready" or "Waiting on Duo"? If the status shows "Waiting on App Provider" then that application type does not yet have Duo frameless support, and can't be used with this SDK.

If your applications indicate they should work with the Universal Prompt but you still receive that 401 message, please get in touch with Duo Support so they can review your account information.

**Comment by nie7321** [comment]
@DuoKristina Aye, our ops folks are talking to support about that. I don't have access to the Duo console, so I can't speak to that stuff. I was just helping them out with an updated OpenAM plugin 😁 

The Java SDK doesn't handle that case very well though -- they had an NPE in their logs and I had to go debug it to figure out what the response was. That could definitely be improved.

Thanks for your help!

**Comment by philsmart** [comment]
This is a bug in the code and should be fixed.

**Comment by AaronAtDuo** [comment]
This should be fixed by https://github.com/duosecurity/duo_universal_java/commit/a49b3a0ab525dd1906d60ebbbe23301b59204d9b from May, as soon as we can get a release done.

**Comment by philsmart** [comment]
Thanks!

**Comment by philliptracy** [comment]
@DuoKristina I'm the Duo admin and work with @nie7321. I created a new Duo application integration to work around the issue - it's identical to the existing one as far as I can see. Both are using Web SDK, and both say "App Update Ready". Well, the new (working) one now says "Waiting on Duo" but you get the idea.  I emailed support yesterday afternoon but haven't heard back yet.

**Comment by DuoKristina** [comment]
@philliptracy Can you respond to the "Thank you for contacting Duo Security Support!" email message you received yesterday with the ikey/client ID of the Web SDK application that isn't working and responds with "Customer does not have frameless access enabled", and we can continue to investigate via that channel instead of here in this unrelated issue?

**Comment by AaronAtDuo** [comment]
@philsmart @nie7321 We just released 1.1.2 with the fix for this issue.  It should be making its way to Maven mirrors over the next few hours.
